### PR TITLE
Allow passing arguments to install scripts

### LINKED ISSUE #1213: pass arguments to install_script
based on installing headers I need to make compatible symlinks for them in $includedir, so I can't have one script, because it doesn't know anything about meson options.

### PR DESCRIPTION
**Pull request by nirbheek**
Closes #1213

Also fixes the usage of install scripts on Windows. Earlier, the test was being entirely skipped on Windows because this was broken.

Also adds a new environment variable `MESON_INSTALL_DESTDIR_PREFIX` which contains both the `DESTDIR` (if it is defined) and the prefix.

The reason why this is needed is the same reason why `destdir_join` is needed in our scripts: `os.path.join` does not work with absolute paths, and it is not trivial to make it work correctly in a cross-platform way.

### CODE REVIEW

File: mesonbuild/scripts/meson_install.py [hunk 1]
```diff
@@ -132,30 +132,19 @@ def install_headers(d):
 def run_install_script(d):
     env = {'MESON_SOURCE_ROOT' : d.source_dir,
            'MESON_BUILD_ROOT' : d.build_dir,
-           'MESON_INSTALL_PREFIX' : d.prefix
+           'MESON_INSTALL_PREFIX' : d.prefix,
```
Comment (ignatenkobrain) [inline mesonbuild/scripts/meson_install.py]: actually I think we should expose here all LIBDIR/BINDIR and all other directories..

**Comment by QuLogic** [comment]
> os.path.join does not work with absolute paths,

Wait, what do you mean by that? `join_paths` is implemented by calling `os.path.join`.

**Comment by nirbheek** [comment]
Yes, and `join_paths` is not the same as [`destdir_join`](https://github.com/mesonbuild/meson/blob/master/mesonbuild/scripts/__init__.py#L15).

**Comment by QuLogic** [comment]
Oh, of course, I misread that as `join_paths`.

**Pull request merged**